### PR TITLE
Cleaned up UserProfile Class

### PR DESCRIPTION
**Pull request by valadas**
- Resolves many stylecop warnings in UserProfile
- Makes constants for retrieving profile properties obsolute and re-implements them as private constants
- Updated some upstream classes to use IPortalSettings instead of PortalSettings
- Implemented automated stylecop fixes on many other files in the same project

### CODE REVIEW

File: DNN Platform/Library/Entities/Users/Profile/ProfilePropertyAccess.cs [hunk 1]
```diff
@@ -49,7 +58,7 @@ public CacheLevel Cacheability
         /// <param name="accessingUser">The accessing user.</param>
         /// <param name="targetUser">The target user.</param>
         /// <returns><c>true</c> if property accessible, otherwise <c>false</c>.</returns>
-        public static bool CheckAccessLevel(PortalSettings portalSettings, ProfilePropertyDefinition property, UserInfo accessingUser, UserInfo targetUser)
+        public static bool CheckAccessLevel(IPortalSettings portalSettings, ProfilePropertyDefinition property, UserInfo accessingUser, UserInfo targetUser)
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/Profile/ProfilePropertyAccess.cs:61]: We need to keep and deprecate the `PortalSettings` overload

File: DNN Platform/Library/Services/Tokens/PropertyAccess/IPropertyAccess.cs [hunk 2]
```diff
@@ -7,10 +7,32 @@ namespace DotNetNuke.Services.Tokens
...
         CacheLevel Cacheability { get; }
-
-        string GetProperty(string propertyName, string format, CultureInfo formatProvider, UserInfo accessingUser, Scope accessLevel, ref bool propertyNotFound);
+
+        /// <summary>
+        /// Gets a property.
+        /// </summary>
+        /// <param name="propertyName">The property name to get.</param>
+        /// <param name="format">The format string to format the property with.</param>
+        /// <param name="formatProvider">The <see cref="CultureInfo"/> format profider to use.</param>
```
Comment (bdukes) [inline DNN Platform/Library/Services/Tokens/PropertyAccess/IPropertyAccess.cs:25]:
```suggestion
        /// <param name="formatProvider">The format provider (e.g. <see cref="CultureInfo"/>) to use.</param>
```

File: DNN Platform/Library/Services/Tokens/PropertyAccess/IPropertyAccess.cs [hunk 3]
```diff
@@ -7,10 +7,32 @@ namespace DotNetNuke.Services.Tokens
...
+        /// </summary>
         CacheLevel Cacheability { get; }
-
-        string GetProperty(string propertyName, string format, CultureInfo formatProvider, UserInfo accessingUser, Scope accessLevel, ref bool propertyNotFound);
+
+        /// <summary>
+        /// Gets a property.
+        /// </summary>
+        /// <param name="propertyName">The property name to get.</param>
+        /// <param name="format">The format string to format the property with.</param>
```
Comment (bdukes) [inline DNN Platform/Library/Services/Tokens/PropertyAccess/IPropertyAccess.cs:24]:
```suggestion
        /// <param name="format">The format string to format the property with, or <c>null</c> to use the default format.</param>
```

**Review by bdukes — COMMENTED** [review]
Thanks!  We just need to add a deprecated `PortalSettings` overload for that `public` method, otherwise it's all good (with a couple of XML doc suggestions).

I've verified this produced [37,378 warnings](https://dev.azure.com/dotnet/DNN/_build/results?buildId=40133&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=a04569be-1b08-5c9a-7e65-a83076350a0c&l=87345) while the previous build produced [37,507 warnings](https://dev.azure.com/dotnet/DNN/_build/results?buildId=40119&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=a04569be-1b08-5c9a-7e65-a83076350a0c&l=87619).

File: DNN Platform/Library/Entities/Users/Profile/UserProfile.cs [hunk 4]
```diff
@@ -3,688 +3,710 @@
...
+    /// <summary>
+    /// The UserProfile class provides a Business Layer entity for the Users Profile.
+    /// </summary>
+    [Serializable]
+    public class UserProfile : IIndexable
+    {
+#pragma warning disable SA1310 // Field names should not contain underscore
+#pragma warning disable SA1600 // Elements should be documented
+        // Name properties
+        [Obsolete("Deprecated in 9.8. Scheduled removal in v10.0.0.")]
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/Profile/UserProfile.cs:32]: What's the suggested replacement if you were using these fields previously?

Comment (bdukes) [reply DNN Platform/Library/Entities/Users/Profile/UserProfile.cs:32]: Maybe just say `Deprectated in v9.8.  Use the properties on this class instead. Scheduled removal in v10.0.0`?

**Comment by valadas** [comment]
>I've verified this produced 37,378 warnings while the previous build produced 37,507 warnings (i.e. this removes 129 warnings).

Yeah, so we are almost done here 🤣

File: DNN Platform/Library/Entities/Users/Profile/ProfilePropertyAccess.cs (hunk 1 above)
Comment (valadas) [reply DNN Platform/Library/Entities/Users/Profile/ProfilePropertyAccess.cs:76]: Makes total sense, implemented that suggestion.

File: DNN Platform/Library/Entities/Users/Profile/UserProfile.cs (hunk 4 above)
Comment (valadas) [reply DNN Platform/Library/Entities/Users/Profile/UserProfile.cs]: I have applied the comments suggestion, I do believe these should have never been made public in the first place. I would gladly be overruled, but I see no reason to expose those as we have much easier way to get to those properties.

**Review by bdukes — APPROVED** [review]
Looks great, thanks!

File: DNN Platform/Library/Entities/Users/Profile/UserProfile.cs (hunk 4 above)
Comment (bdukes) [reply DNN Platform/Library/Entities/Users/Profile/UserProfile.cs]: Agreed, those shouldn't need to be exposed.

**Comment by bdukes** [comment]
I did some analysis on the build log, and if we ignore the SA16xx documentation warnings and the CS0618 obsolete warnings, we're almost there with only a little over 10,000 warnings left.

I am this close GIF

**Review by mitchelsellers — APPROVED** [review]

**Pull request merged**